src/python/src/grpc/_links/_transmission_test.py

```diff
@@ -94,7 +94,12 @@ def create_service_completion(self):
     return _intermediary_low.Code.OK, 'An exuberant test "details" message!'
 
   def assertMetadataEqual(self, original_metadata, transmitted_metadata):
-    self.assertSequenceEqual(original_metadata, transmitted_metadata)
+    # we need to filter out any additional metadata added in transmitted_metadata
+    # since implementations are allowed to add to what is sent (in any position)
+    keys, _ = zip(*original_metadata)
```

Comment:
I'm not yet convinced that this check is correct. Consider the application supplying metadata

"a": "1"
"b": "2"

and the runtime adding additional metadata

"b": "3".

Won't the assertSequenceEqual call raise an exception in that case? Shouldn't this method instead return without raising an exception?

Am I right in thinking that the logic for this method should be "verify that the key-value pairs in original_metadata are a subset of the key-value pairs in transmitted_metadata and that they appear in the same order relative to one another notwithstanding any other pairs inserted among them"?